src/cow-react/modules/swap/state/EthFlow/updaters/EthFlowSlippageUpdater.tsx

```diff
@@ -1,70 +1,120 @@
-    if (currentSlippage !== 'auto' && currentSlippage.equalTo(ETH_FLOW_SLIPPAGE)) {
-      setUserSlippageTolerance(previousSlippage)
+    const { ethFlow } = _loadSlippage()
+
+    if (
+      isEthFlow &&
+      ethFlow &&
+      ((currentSlippage === 'auto' && ethFlow !== 'auto') ||
+        (currentSlippage instanceof Percent && ethFlow instanceof Percent && !currentSlippage.equalTo(ethFlow)))
+    ) {
+      // If ethFlow and there's an enflow stored which is not auto
+      setUserSlippageTolerance(ethFlow)
     }
 
-    setMounted(true)
     // we only want this on mount
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  // change slippage set to 2% when detected eth swap and option is set to use native flow
+  // ref used to track when EthFlow is disabled
+  const wasEthFlowActive = useRef(false)
+
   useEffect(() => {
     if (isEthFlow) {
-      // save the previous slippage to set back if users switches out of native swap
-      _saveSlippage(currentSlippage)
+      // Load what's stored
+      const { regular, ethFlow } = _loadSlippage()
+
+      // Set the flag
+      wasEthFlowActive.current = true
+
+      if (
+        currentSlippage === 'auto' ||
+        (!currentSlippage.greaterThan(ETH_FLOW_SLIPPAGE) && !currentSlippage.equalTo(ETH_FLOW_SLIPPAGE))
+      ) {
+        // If current slippage is auto or if it's smaller than ETH flow slippage, update it
 
-      setUserSlippageTolerance(ETH_FLOW_SLIPPAGE)
-      // only after it's run the effect once
-    } else if (mounted) {
+        // If the former ethFlow slippage was saved, use that. Otherwise pick the minimum
+        const newSlippage =
+          ethFlow !== 'auto' && ethFlow && ethFlow.greaterThan(ETH_FLOW_SLIPPAGE) ? ethFlow : ETH_FLOW_SLIPPAGE
+
+        // Update the global state
+        setUserSlippageTolerance(newSlippage)
+
+        // Update local storage
+        _saveSlippage({ regular: regular || currentSlippage, ethFlow: newSlippage })
+      } else {
+        // If current slippage is NOT auto and it's greater than minimum, store that locally
+        _saveSlippage({ regular, ethFlow: currentSlippage })
+      }
+    } else if (wasEthFlowActive.current) {
+      // Only when disabling EthFlow, reset to previous regular value
       _resetSlippage(setUserSlippageTolerance)
+      // Disable the flag
+      wasEthFlowActive.current = false
     }
-    // we only want to depend on isEthFlow
-    // to avoid re-renders
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [setUserSlippageTolerance, isEthFlow])
+  }, [setUserSlippageTolerance, isEthFlow, currentSlippage])
 
   return null
 }
 
-type SerializedSlippage = 'auto' | [string, string]
+function _saveSlippage(slippageSettings: SlippageSettings): void {
+  setJsonToLocalStorage(LOCAL_STORAGE_KEY, _serialize(slippageSettings))
+}
+
+function _serialize({ regular, ethFlow }: SlippageSettings): SerializedSlippageSettings {
```

Comment:
Just reflections:
We could use `atomWithStorage` and a custom getter for deserialization.

Reply:
Indeed we could!
I just decided not to change what David started, but maybe I should.
Will consider that for refactoring as good to have

Reply:
https://github.com/cowprotocol/cowswap/issues/1485